### DISCUSSION TITLE
Trigger form input validation only on submit

### DISCUSSION THREAD

**Discussion by guizoxxv** [element-plus/element-plus#2528] [Q&A] [answered]
Hi. I would like to execute front end validation only on form submit.

Right now the `trigger` attribute of `rules` accept `blur` or `change`. I tried to remove the `trigger` attribute but it seems that it defaults to the `blur` behavior.

Is there a way to achieve this? If not I think it would be nice to have this option. Thank you.

**Comment by mPyth** [comment]
I have the same problem, question is added a few minutes ago in #8666 with demo example.

**Comment by vasylpylypets** [comment]
```
<el-button type="primary" @click="submitForm(ruleFormRef)">Create</el-button>

const submitForm = (formEl) => {
    if (!formEl) return;
    formEl.validateField([], valid => {
        if (valid) {
            console.log('submit!');
        } else {
            console.log('error submit!', fields);
        }
    });
};
```

formEl.validateField(arrayOfFields, callback) - this method triggers validation
arrayOfFields - specific list of fields to validate or empty array as "all fields"

**Reply by guizoxxv** [reply]
Thanks @vasylpylypets, but this does not prevents the validation from being triggered on blur, which seems to be default. [Here](https://codepen.io/guizo/pen/ZEMGKVR) is a demo.

**Comment by chenxch** [comment, marked as answer]
We provide the `validate-event` attribute in the component, you can set it to false, and the validation will not be triggered on the input.
[demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtZm9ybVxuICAgIHJlZj1cInJ1bGVGb3JtUmVmXCJcbiAgICA6bW9kZWw9XCJydWxlRm9ybVwiXG4gICAgOnJ1bGVzPVwicnVsZXNcIlxuICAgIGxhYmVsLXdpZHRoPVwiMTIwcHhcIlxuICAgIGNsYXNzPVwiZGVtby1ydWxlRm9ybVwiXG4gICAgOnNpemU9XCJmb3JtU2l6ZVwiXG4gICAgc3RhdHVzLWljb25cbiAgPlxuICAgIDxlbC1mb3JtLWl0ZW0gbGFiZWw9XCJBY3Rpdml0eSBuYW1lXCIgcHJvcD1cIm5hbWVcIj5cbiAgICAgIDxlbC1pbnB1dCB2LW1vZGVsPVwicnVsZUZvcm0ubmFtZVwiIDp2YWxpZGF0ZS1ldmVudD1cImZhbHNlXCIvPlxuICAgIDwvZWwtZm9ybS1pdGVtPlxuICAgIDxlbC1mb3JtLWl0ZW0+XG4gICAgICA8ZWwtYnV0dG9uIHR5cGU9XCJwcmltYXJ5XCIgQGNsaWNrPVwic3VibWl0Rm9ybShydWxlRm9ybVJlZilcIj5cbiAgICAgICAgQ3JlYXRlXG4gICAgICA8L2VsLWJ1dHRvbj5cbiAgICAgIDxlbC1idXR0b24gQGNsaWNrPVwicmVzZXRGb3JtKHJ1bGVGb3JtUmVmKVwiPlJlc2V0PC9lbC1idXR0b24+XG4gICAgPC9lbC1mb3JtLWl0ZW0+XG4gIDwvZWwtZm9ybT5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWFjdGl2ZSwgcmVmIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHR5cGUgeyBGb3JtSW5zdGFuY2UsIEZvcm1SdWxlcyB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuY29uc3QgZm9ybVNpemUgPSByZWYoJ2RlZmF1bHQnKVxuY29uc3QgcnVsZUZvcm1SZWYgPSByZWY8Rm9ybUluc3RhbmNlPigpXG5jb25zdCBydWxlRm9ybSA9IHJlYWN0aXZlKHtcbiAgbmFtZTogJ0hlbGxvJyxcbiAgcmVnaW9uOiAnJyxcbiAgY291bnQ6ICcnLFxuICBkYXRlMTogJycsXG4gIGRhdGUyOiAnJyxcbiAgZGVsaXZlcnk6IGZhbHNlLFxuICB0eXBlOiBbXSxcbiAgcmVzb3VyY2U6ICcnLFxuICBkZXNjOiAnJyxcbn0pXG5cbmNvbnN0IHJ1bGVzID0gcmVhY3RpdmU8Rm9ybVJ1bGVzPih7XG4gIG5hbWU6IFtcbiAgICB7IHJlcXVpcmVkOiB0cnVlLCBtZXNzYWdlOiAnUGxlYXNlIGlucHV0IEFjdGl2aXR5IG5hbWUnLCB0cmlnZ2VyOiAnYmx1cicgfSxcbiAgICB7IG1pbjogMywgbWF4OiA1LCBtZXNzYWdlOiAnTGVuZ3RoIHNob3VsZCBiZSAzIHRvIDUnLCB0cmlnZ2VyOiAnYmx1cicgfSxcbiAgXSxcbiAgcmVnaW9uOiBbXG4gICAge1xuICAgICAgcmVxdWlyZWQ6IHRydWUsXG4gICAgICBtZXNzYWdlOiAnUGxlYXNlIHNlbGVjdCBBY3Rpdml0eSB6b25lJyxcbiAgICAgIHRyaWdnZXI6ICdjaGFuZ2UnLFxuICAgIH0sXG4gIF0sXG4gIGNvdW50OiBbXG4gICAge1xuICAgICAgcmVxdWlyZWQ6IHRydWUsXG4gICAgICBtZXNzYWdlOiAnUGxlYXNlIHNlbGVjdCBBY3Rpdml0eSBjb3VudCcsXG4gICAgICB0cmlnZ2VyOiAnY2hhbmdlJyxcbiAgICB9LFxuICBdLFxuICBkYXRlMTogW1xuICAgIHtcbiAgICAgIHR5cGU6ICdkYXRlJyxcbiAgICAgIHJlcXVpcmVkOiB0cnVlLFxuICAgICAgbWVzc2FnZTogJ1BsZWFzZSBwaWNrIGEgZGF0ZScsXG4gICAgICB0cmlnZ2VyOiAnY2hhbmdlJyxcbiAgICB9LFxuICBdLFxuICBkYXRlMjogW1xuICAgIHtcbiAgICAgIHR5cGU6ICdkYXRlJyxcbiAgICAgIHJlcXVpcmVkOiB0cnVlLFxuICAgICAgbWVzc2FnZTogJ1BsZWFzZSBwaWNrIGEgdGltZScsXG4gICAgICB0cmlnZ2VyOiAnY2hhbmdlJyxcbiAgICB9LFxuICBdLFxuICB0eXBlOiBbXG4gICAge1xuICAgICAgdHlwZTogJ2FycmF5JyxcbiAgICAgIHJlcXVpcmVkOiB0cnVlLFxuICAgICAgbWVzc2FnZTogJ1BsZWFzZSBzZWxlY3QgYXQgbGVhc3Qgb25lIGFjdGl2aXR5IHR5cGUnLFxuICAgICAgdHJpZ2dlcjogJ2NoYW5nZScsXG4gICAgfSxcbiAgXSxcbiAgcmVzb3VyY2U6IFtcbiAgICB7XG4gICAgICByZXF1aXJlZDogdHJ1ZSxcbiAgICAgIG1lc3NhZ2U6ICdQbGVhc2Ugc2VsZWN0IGFjdGl2aXR5IHJlc291cmNlJyxcbiAgICAgIHRyaWdnZXI6ICdjaGFuZ2UnLFxuICAgIH0sXG4gIF0sXG4gIGRlc2M6IFtcbiAgICB7IHJlcXVpcmVkOiB0cnVlLCBtZXNzYWdlOiAnUGxlYXNlIGlucHV0IGFjdGl2aXR5IGZvcm0nLCB0cmlnZ2VyOiAnYmx1cicgfSxcbiAgXSxcbn0pXG5cbmNvbnN0IHN1Ym1pdEZvcm0gPSBhc3luYyAoZm9ybUVsOiBGb3JtSW5zdGFuY2UgfCB1bmRlZmluZWQpID0+IHtcbiAgaWYgKCFmb3JtRWwpIHJldHVyblxuICBhd2FpdCBmb3JtRWwudmFsaWRhdGUoKHZhbGlkLCBmaWVsZHMpID0+IHtcbiAgICBpZiAodmFsaWQpIHtcbiAgICAgIGNvbnNvbGUubG9nKCdzdWJtaXQhJylcbiAgICB9IGVsc2Uge1xuICAgICAgY29uc29sZS5sb2coJ2Vycm9yIHN1Ym1pdCEnLCBmaWVsZHMpXG4gICAgfVxuICB9KVxufVxuXG5jb25zdCByZXNldEZvcm0gPSAoZm9ybUVsOiBGb3JtSW5zdGFuY2UgfCB1bmRlZmluZWQpID0+IHtcbiAgaWYgKCFmb3JtRWwpIHJldHVyblxuICBmb3JtRWwucmVzZXRGaWVsZHMoKVxufVxuXG5jb25zdCBvcHRpb25zID0gQXJyYXkuZnJvbSh7IGxlbmd0aDogMTAwMDAgfSkubWFwKChfLCBpZHgpID0+ICh7XG4gIHZhbHVlOiBgJHtpZHggKyAxfWAsXG4gIGxhYmVsOiBgJHtpZHggKyAxfWAsXG59KSlcbjwvc2NyaXB0PlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

**Reply by guizoxxv** [reply]
Thanks. I was focused on the form and form-item API and didn't saw this on the input API. I opened a [PR](https://github.com/element-plus/element-plus/pull/11745) to add a tip about this on the form documentation page.

**Comment by mPyth** [comment]
`validate-event` works great. Thanks for your time and for demo.
